backend/netstorage/netstorage.go

```diff
@@ -0,0 +1,1369 @@
+// Package netstorage provides an interface to Akamai NetStorage API
+package netstorage
+
+import (
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/xml"
+	"io"
+	"io/ioutil"
+	"math/rand"
+	"net/http"
+	"net/url"
+	"runtime/debug"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/rclone/rclone/fs"
+	"github.com/rclone/rclone/fs/config/configmap"
+	"github.com/rclone/rclone/fs/config/configstruct"
+	"github.com/rclone/rclone/fs/config/obscure"
+	"github.com/rclone/rclone/fs/fshttp"
+	"github.com/rclone/rclone/fs/hash"
+	"github.com/rclone/rclone/fs/walk"
+	"github.com/rclone/rclone/lib/pacer"
+	"github.com/rclone/rclone/lib/rest"
+	"golang.org/x/net/html/charset"
+)
+
+// Constants
+const (
+	timeFormatIn  = time.UnixDate
+	minSleep      = 10 * time.Millisecond
+	maxSleep      = 2 * time.Second
+	decayConstant = 2 // bigger for slower decay, exponential
+)
+
+func init() {
+	fsi := &fs.RegInfo{
+		Name:        "netstorage",
+		Description: "Akamai NetStorage",
+		NewFs:       NewFs,
+		CommandHelp: commandHelp,
+		Options: []fs.Option{{
+			Name: "protocol",
+			Help: "Select between HTTP or HTTPS protocol. Most users should choose HTTPS, which is the default. HTTP is provided primarily for debugging purposes.",
+			Examples: []fs.OptionExample{{
+				Value: "http",
+				Help:  "HTTP protocol",
+			}, {
+				Value: "https",
+				Help:  "HTTPS protocol",
+			}},
+			Required: true,
+		}, {
+			Name:     "host",
+			Help:     "domain+path of netstorage host to connect to. Format should be <domain>/<internal folders>",
+			Required: true,
+		}, {
+			Name:     "account",
+			Help:     "set the netstorage account name",
+			Required: true,
+		}, {
+			Name:       "secret",
+			Help:       "set the netstorage account secret/G2O key which will be used for authentication purposes. please choose the 'y' option to set your own password then enter your secret.",
```

Comment:
You want to set IsPassword if the key is the kind of thing someone could "eyedrop" lets say "mypassword" - if it is a long hex string then we don't normally encode these in rclone as they can't be memorised easily.